collectors/mesos/agent/agent.go

```diff
@@ -150,6 +150,16 @@ func getFrameworkInfoByFrameworkID(frameworkID string, frameworks []frameworkInf
 	return frameworkInfo{}, false
 }
 
+// getExecutorInfoByExecutorID returns the ExecutorInfo struct given its ID.
+func getExecutorInfoByExecutorID(executorID string, executors []executorInfo) (executorInfo, bool) {
```

Comment:
I think this method could just return a `*executorInfo` since the boolean value only signals whether or not you found it.  Nil/Non-Nil would do the same thing as well as prevent an unnecessary allocation if you could not find the executor for the given ID.